### ISSUE TITLE
Conflict between '@typescript-eslint/no-extra-parens' and 'no-confusing-arrow'

### ISSUE THREAD

**Issue by adamhill86** [perfective/eslint-config#1]
## 🤔 Problem:

With the rules are configured, the following line throws the warning `Unnecessary parentheses around expression eslint(@typescript-eslint/no-extra-parens)`:

```TypeScript
return allPostsData.sort((a, b) => (a.date < b.date ? 1 : -1));
```

Fixing the warning by removing the parentheses around the ternary expression results in a different warning, `Arrow function used ambiguously with a conditional expression eslint(no-confusing-arrow)`:

```TypeScript
return allPostsData.sort((a, b) => a.date < b.date ? 1 : -1);
```

## 💡 Proposed solution:

These rules can live in harmony by making a single change:

```TypeScript
        '@typescript-eslint/no-extra-parens': ['warn', 'all', {
            // eslint-disable-next-line @typescript-eslint/naming-convention -- external property
            ignoreJSX: 'all',
            // Nested binary expressions may be hard to break down into named functions
            nestedBinaryExpressions: false,
            enforceForArrowConditionals: false, // <-- this needs to be set to false
        }],
```

Per discussion in this thread: https://github.com/eslint/eslint/issues/6196#issuecomment-333344466

**Comment by amikheychik** [comment]
@adamhill86 makes sense, thank you for finding a solution. I'll add it as a patch.

**Comment by amikheychik** [comment]
This issue is resolved in the `@perfective/eslint-config@0.15.1`

**Comment by TokugawaTakeshi** [comment]
@amikheychik,

> This issue is resolved in the `@perfective/eslint-config@0.15.1`

I am sorry, but it's unclear HOW this problem has been resolved.
Could you please to post the right combination of options for `@typescript-eslint/no-extra-parens` and `no-confusing-arrow`?

**Comment by amikheychik** [comment]
@TokugawaTakeshi I've just applied the fix suggested by @adamhill86 and turned off `enforceForArrowConditionals` in the `no-extra-parens` rule (see https://github.com/perfective/eslint-config/commit/8a745ddd18d67032c419445e301cf7d8e1b6fa66)
